jetson/nav/README.md

```diff
@@ -0,0 +1,84 @@
+
+#### `stateMachine.cpp`
+This file contains implementations of the stateMachine object’s member functions, including the `run()` function, which executes the logic for switching between navigation states and calling the functions to run in each state.
+
+#### `rover.cpp`
+This file defines the rover and rover status objects. The rover object is used throughout the codebase to interact with real-life capabilities of the rover. Notably, the object contains functions like `drive()` and `turn()`. The rover status object/class is nested in the rover class, and it contains information about the current state of the rover and relevant features like targets and obstacles. Most variables in the rover status are populated from LCM messages.
+
+---
+
+<!----------------------------- Gate Search ----------------------------->
+## Gate Search (`gate_search/` folder)
+
+#### `diamondGateSearch.cpp`
+This file creates the search waypoints in the shape of a diamond for completing a search for the second gate post.
+
+#### `gateStateMachine.cpp`
+Defines gate search/traversal states and functions.
+
+
+---
+
+<!----------------------------- Obstacle Avoidance ----------------------------->
+## Obstacle Avoidance
+
+#### `obstacleAvoidanceStateMachine.cpp`
+Defines an obstacle avoidance state machine with minimal functionality, intended to be a parent class for different types of obstacle avoidance strategies
+
+#### `simpleAvoidance.cpp`
+This contains what is currently our only implementation of obstacle avoidance behavior. Inherited from the obstacle state machine, it is a very simple algorithm and just drops a waypoint at the front of the queue, with a position at a safe location away from the obstacle, for the rover to drive to before continuing to its previous destination. If we developed another behavior algorithm, we would make another obstacle state machine child class like simpleAvoidance.
+
+
+---
+
+<!----------------------------- Search ----------------------------->
+## Search
+Similar to the `gate_search/` folder, this folder for search logic contains a `searchStateMachine` object and files to define the waypoints for different types of searches. First we follow a square spiral outwards with points generated in spiralOutSearch.cpp, then if the search completes and the target is not found, we will move onto trying the lawnmower search and the spiral in search.
+
+
+---
+
+<!----------------------------- Vars and Utils ----------------------------->
+## Variables and Utilities
+
+#### Nav State
+The nav state specifies what state of the state machine we are in. It is implemented as an enum (a C++ type) where named states are associated with a number behind the scenes.
+
+#### Auton State
+This is just a boolean hidden as a type called AutonState and it tells us if Auton is on or off. It is read from LCMs published by the GUI or simulator.
+
+##### `utilities.cpp`
+Contains functions used commonly throughout auton code. 
+
+
```

Comment:
also how to run this code base, or like what is required to including for testing on rover and in simulator

Reply:
Added in simulator commands, will add rover commands